### PR TITLE
AnP Reprocess

### PR DESCRIPTION
**Pull request by ebelter**
Allow the reprocessing of select instrument data for an AnP.

### CODE REVIEW

**Comment by ebelter** [comment]
The test worked, put in pratcie, the command didn't. Used instdata ids to get bridges.

File: lib/perl/Genome/Config/AnalysisProject/Command/Reprocess.pm [hunk 1]
```diff
@@ -35,12 +42,23 @@ sub valid_statuses {
...
 }
 
+sub _resolve_analysis_project_bridges {
+    my $self = shift;
+
+    return $self->analysis_project->analysis_project_bridges if not $self->instrument_data;
+
+    return Genome::Config::AnalysisProject::InstrumentDataBridge->get(
+        analysis_project => $self->analysis_project,
+        'instrument_data.id' => [ map { $_->id}  $self->instrument_data ],
```
Comment (tmooney) [inline lib/perl/Genome/Config/AnalysisProject/Command/Reprocess.pm]: `instrument_data_id` is a direct property of the `InstrumentDataBridge`, so we don't need to use the `.` syntax here.

Comment (ebelter) [inline lib/perl/Genome/Config/AnalysisProject/Command/Reprocess.pm]: @tmooney But I WANT to use it :man_with_gua_pi_mao:

Comment (tmooney) [inline lib/perl/Genome/Config/AnalysisProject/Command/Reprocess.pm]: :smile:, but it does an extra join!

Another option more parallel to the "all" case might be:
```perl
return $self->analysis_project->analysis_project_bridges(instrument_data_id => [map $_->id, $self->instrument_data]);
```

**Pull request merged**